modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusionComplete.C

```diff
@@ -0,0 +1,204 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "INSFVMomentumDiffusionComplete.h"
+#include "INSFVRhieChowInterpolator.h"
+#include "NS.h"
+#include "SystemBase.h"
+#include "RelationshipManager.h"
+#include "Factory.h"
+
+registerMooseObject("NavierStokesApp", INSFVMomentumDiffusionComplete);
+
+InputParameters
+INSFVMomentumDiffusionComplete::validParams()
+{
+  auto params = INSFVMomentumDiffusion::validParams();
+  params.addClassDescription(
+      "Implements the full Laplace form of the viscous stress in the Navier-Stokes equation.");
+  params.addRequiredCoupledVar("u", "The velocity in the x direction.");
+  params.addCoupledVar("v", "The velocity in the y direction.");
+  params.addCoupledVar("w", "The velocity in the z direction.");
+  params.addParam<bool>("harmonic_interpolation", true, "Whether to use harmonic interpolation for viscosity.");
+  params.set<unsigned short>("ghost_layers") = 2;
+  return params;
+}
+
+INSFVMomentumDiffusionComplete::INSFVMomentumDiffusionComplete(const InputParameters & params)
+  : INSFVMomentumDiffusion(params),
+    _dim(_subproblem.mesh().dimension()),
+    _u_var(dynamic_cast<const INSFVVelocityVariable *>(getFieldVar("u", 0))),
+    _v_var(params.isParamValid("v")
+               ? dynamic_cast<const INSFVVelocityVariable *>(getFieldVar("v", 0))
+               : nullptr),
+    _w_var(params.isParamValid("w")
+               ? dynamic_cast<const INSFVVelocityVariable *>(getFieldVar("w", 0))
+               : nullptr),
+    _harmonic_interpolation(getParam<bool>("harmonic_interpolation"))
+{
+  if ((_var.faceInterpolationMethod() == Moose::FV::InterpMethod::SkewCorrectedAverage) &&
+      (_tid == 0))
+  {
+    auto & factory = _app.getFactory();
```

Comment:
This has changed lately, check `FVDiffusion` for an example. It will be something like this:

```
if ((_var.faceInterpolationMethod() == Moose::FV::InterpMethod::SkewCorrectedAverage) &&
      (_tid == 0))
    adjustRMGhostLayers(std::max((unsigned short)(3), _pars.get<unsigned short>("ghost_layers")));
```